fairmq/shmem/Common.h

```diff
@@ -69,13 +176,24 @@ struct RegionInfo
         , fCreationFlags(flags)
         , fUserFlags(userFlags)
         , fSize(size)
+        , fRCSegmentSize(10000000)
+        , fDestroyed(false)
+    {}
```

Comment:
Could be shortened with a delegating ctor (same with the one ctor above):

```cpp
    RegionInfo(const char* path, int flags, uint64_t userFlags, uint64_t size, const VoidAlloc& alloc)
        : RegionInfo(path, flags, userFlags, size, 10000000, alloc)
    {}
```

And since `10000000` is used twice, I suggest to deduplicate, e.g.
```cpp
    constexpr uint64_t DefaultRcSegmentSize = 10000000;    
```

Reply:
alternative would be to use default member initializers:
```cpp
struct RegionInfo
{
    RegionInfo(const VoidAlloc& alloc)
        : fPath("", alloc)

    RegionInfo(const char* path, int flags, uint64_t userFlags, uint64_t size, const VoidAlloc& alloc)
        : fPath(path, alloc)
        , fCreationFlags(flags)
        , fUserFlags(userFlags)
        , fSize(size)
    {}

    RegionInfo(const char* path, int flags, uint64_t userFlags, uint64_t size, uint64_t rcSegmentSize, const VoidAlloc& alloc)
        : fPath(path, alloc)
        , fCreationFlags(flags)
        , fUserFlags(userFlags)
        , fSize(size)
        , fRCSegmentSize(rcSegmentSize)
    {}

    Str fPath;
    int fCreationFlags{0}; // = 0; should work as well
    uint64_t fUserFlags{0};
    uint64_t fSize{0};
    uint64_t fRCSegmentSize{10000000};
    bool fDestroyed{false};
};
```